### ISSUE TITLE
[a11y]: Screen readers don't read listed items in Dropdown menu when id specified

### ISSUE THREAD

**Issue by denbon05** [carbon-design-system/carbon#17915]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.62.0

### React version

18.2.0

### Automated testing tool and ruleset

Exploratory testing

### Assistive technology

https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn

### Description

Screen reads I tried seems unable to read Dropdown items.
[It works fine in the default sample](https://react.carbondesignsystem.com/?path=/story/components-dropdown--default), but not with a collection whose items contain an `id` property

### WCAG 2.1 Violation

WCAG 2.1

### Reproduction/example

https://stackblitz.com/edit/github-jru9nu?file=src%2FApp.jsx,package.json

### Steps to reproduce

1. Create a simple React component:

```jsx
import { Dropdown } from '@carbon/react';
import React from 'react';

export default function App() {
  const items = [
    { name: '111', id: '1' },
    { name: '222', id: '2' },
    { name: '333', id: '3' },
    { name: '444', id: '4' },
  ];

  return (
    <div>
      <Dropdown
        id="default"
        label="Dropdown"
        items={items}
        initialSelectedItem={items[1]}
        titleText="Dropdown label"
        helperText="This is some helper text"
        itemToString={(item) => (item ? item.name : '')}
      />
    </div>
  );
}

```

2. [Add extension to the browser](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn)
3. Try to use screen reader on dropdown menu using arrows to navigate through the list

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by denbon05** [comment]
Closing, as it is more specific description [of a known issue](https://github.com/carbon-design-system/carbon/issues/17582)
